dsl_auto_type/src/auto_type/expression_type_inference.rs

```diff
@@ -302,6 +302,54 @@ impl TypeInferrer<'_> {
                     }
                 }
             }
+            (syn::Expr::Binary(binary_expression), None) => {
+                let op_span = Span::mixed_site().located_at(binary_expression.op.span());
+                let trait_name = match binary_expression.op {
+                    syn::BinOp::Add(_) => "Add",
+                    syn::BinOp::Sub(_) => "Sub",
+                    syn::BinOp::Mul(_) => "Mul",
+                    syn::BinOp::Div(_) => "Div",
+                    syn::BinOp::Rem(_) => "Rem",
+                    syn::BinOp::And(_) => return Ok(parse_quote!(bool)),
+                    syn::BinOp::Or(_) => return Ok(parse_quote!(bool)),
+                    syn::BinOp::BitXor(_) => "BitXor",
+                    syn::BinOp::BitAnd(_) => "BitAnd",
+                    syn::BinOp::BitOr(_) => "BitOr",
+                    syn::BinOp::Shl(_) => "Shl",
+                    syn::BinOp::Shr(_) => "Shr",
+                    syn::BinOp::Eq(_) => "PartialEq",
```

Comment:
I think that case won't work. [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) is neither a member of `std::ops` nor has it an associated type `Output`. Maybe we can just mark it as unsupported for now?

Reply:
Ah tight this one should have been amon the "always boolean". 👍